style.md

```diff
@@ -1557,7 +1557,237 @@ type Client struct {
 </td></tr>
 </tbody></table>
 
-### Use Field Names to initialize Structs
+As a general rule of thumb, embedding should not make types more difficult to
+use, and should be viewed as an "is-a" declaration: the embedding (outer) type
+can function as the embedded (inner) type, and therefore "is a" form of the
+inner type. A type embedding a `sync.Mutex` declares to users that it "is a"
+mutex; a type embedding a `bytes.Buffer` declares that it "is a" buffer.
+
+Embedded types' methods or fields should never be shadowed by first-class
+methods or fields defined as part of the embedding type, and embedding should
+never be used for callsite brevity or other cosmetic convenience.
+
+When embedding, visibility and type semantics should be taken into account:
+because exported fields and methods of an embedded type will also be exported
+by the embedding type, careless use can bleed implementation details through
+types' API boundaries, expose internal type faculties that can be used in an
+unsafe or unknown way, or change the semantics of the embedding type altogether.
+
+All of the guidance in this section applies primarily to embedding types that
+are either (a) exported or (b) unexported but conventionally satisfy public
+interfaces, especially if the embedding type can be trivially type-converted to
+an interface that the embedded type satsfies.
+
+#### Embedding Should Not Affect Useful Zero Values
+
+If a given embedding (outer) type has a useful zero value (e.g. `bytes.Buffer`),
+an embedded type should not cause the outer type to no longer be useful as a
+zero value. Similarly, if an outer type can be used as a value type or otherwise
+be trivially struct-initialized, embedded types should be optional or
+lazily-populated in order to minimize initialization complexiy of
+secondary (embedded) types or fields.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+type Book struct {
+    io.ReadWriter
+
+    // other fields
+}
+
+// later
+
+var b Book
+b.Read(...)  // panic
+b.String()   // panic
+b.Write(...) // panic
+```
+
+</td><td>
+
+```go
+type Book struct {
+    bytes.Buffer
+
+    // other fields
+}
+
+// later
+
+var b Book
+b.Read(...)  // ok
+b.String()   // ok
+b.Write(...) // ok
+```
+
+</td></tr>
+</tbody></table>
+
+#### Embedding Should Not Change Type Semantics
+
+An embedded (inner) type should not add semantically unrelated or extraneous
+methods or fields to the embedding (outer) type's exported API, nor should it
+add methods or fields that imply that the type behaves other than explicitly
+defined. This applies both to newly-available (via embedding) and "shadowed"
+methods and fields.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+type Car struct {
```

Comment:
i think this example has a lot going on:
 - multiple embeds, neither of which is related and leaks unnecessary details
 - shadowing with a different intent

Can we distill this a little -- maybe just a `Car` type with a mutex?